File: Tienda_Java/src/Cliente.java
import java.util.*;
public class Cliente {
    String nombre;
    private ArrayList<Producto> compras = new ArrayList<Producto>();;

    Cliente(String nombre) {
        this.nombre = nombre;
        this.compras = new ArrayList<>();
    }

    void comprar(Producto producto) {
        compras.add(producto);
        Producto.eliminarStock();
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public List<Producto> getCompras() {
        return compras;
    }

    public void setCompras(ArrayList<Producto> compras) {
        this.compras = compras;
    }
}

File: Tienda_PY/Cliente.py
class Cliente:
  
  def __init__(self, nombre):
    self.nombre = nombre
    self.compras = []

  def comprar(self, producto):
    self.compras.append(producto)
    producto.eliminar_stock()

  def get_nombre(self):
    return self.nombre

  def set_nombre(self, nombre):
    self.nombre = nombre

  def get_compras(self):
    return self.compras

  def set_compras(self, compras):
    self.compras = compras